src/ed25519.rs

```diff
@@ -8,24 +8,78 @@
 //! # Example (with ed25519-dalek)
 //!
 //! ```nobuild
-//! extern crate signatory;
-//! extern crate signatory_dalek; // or another Ed25519 provider
+//! # Sample Test case for Signing and Verifying a digital signature using the *signatory* crate
 //!
-//! use signatory::ed25519;
+//! The Signatory crates provides a unified API for accessing various crypto
+//! providers for digital signatures.
+//!
+//! Signatory provides a thread-safe and object-safe API and implements providers
+//! for many popular Rust crates, including [ed25519‑dalek], [*ring*], [secp256k1],
+//! and [sodiumoxide].
+//!
+//! This example uses a unique private key generated by the ed25519 crate, signs a
+//! piece of text using the Signer function from the signature crate, and then
+//! verifies the signature using the corresponding public key.
+//!
+//! The second test use a deterministic key generated from a user supplied *Secret*
+//! and a *Salt*.
+//! ```
+//! [dependencies]
+//! base64 = "*"
+//! signatory-dalek = "0.20.0"
+//! signatory = "0.20.0"
+//!
+//! sha2 = "0.9.0"
+//! hmac = "0.8.0"
+//! ```
+//!
+//! ```
 //! use signatory_dalek::{Ed25519Signer, Ed25519Verifier};
 //!
-//! // Create a private key (a.k.a. a "seed") and use it to generate a signature
-//! let seed = ed25519::Seed::generate();
-//! let signer = Ed25519Signer::from(&seed);
-//! let msg = "How are you? Fine, thank you.";
+//! use signatory::{
+//!     ed25519, public_key::PublicKeyed, signature::Signature, signature::Signer, signature::Verifier,
+//! };
```

Comment:
Can you group the vicarious imports from `signature`?

```suggestion
//! use signatory::{
//!     ed25519, public_key::PublicKeyed,
//!     signature::{Signature, Signer, Verifier},
//! };
```

Reply:
Done. I didn't know I could group import like this. Thanks.